### ISSUE TITLE
Pubsub 2.x: default scope not set in the Credentials class

### ISSUE THREAD

**Issue by qnighy** [googleapis/google-cloud-ruby#8610]
#### Environment details

  - OS: Linux, Mac OS
  - Ruby version: Ruby 2.6.6, 2.7.1
  - Gem name and version: google-cloud-pubsub 2.x only

#### Steps to reproduce

```ruby
Google::Cloud::Pubsub::Credentials.new(keyfile)
```

leads to the following error:

```
/Users/qnighy/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/signet-0.14.0/lib/signet/oauth_2/client.rb:1039:in `fetch_access_token': Authorization failed.  Server message: (Signet::AuthorizationError)
{"error":"invalid_scope","error_description":"Invalid OAuth scope or ID token audience provided."}
```

This is because the following value is `nil`.

```ruby
Google::Cloud::Pubsub::Credentials.scope
```

In google-cloud-pubsub 1.x, the method above returns a valid array of scopes.

The regression seems to be caused by the change from

> ```ruby
>           SCOPE = ["https://www.googleapis.com/auth/pubsub"].freeze
> ```

https://github.com/googleapis/google-cloud-ruby/blob/google-cloud-pubsub/v1.10.0/google-cloud-pubsub/lib/google/cloud/pubsub/v1/credentials.rb#L23

to

> ```ruby
>             self.scope = [
>               "https://www.googleapis.com/auth/cloud-platform",
>               "https://www.googleapis.com/auth/pubsub"
>             ]
> ```

https://github.com/googleapis/google-cloud-ruby/blob/google-cloud-pubsub/v2.3.1/google-cloud-pubsub-v1/lib/google/cloud/pubsub/v1/publisher/credentials.rb#L28-L31

where the value set by `scope=` is not inherited to the subclass `Google::Cloud::PubSub::Credentials`, unlike the one by `SCOPE = `.

**Comment by quartzmo** [comment]
@qnighy Thank you for reporting this issue!

**Comment by quartzmo** [comment]
I have reproduced the observed error. And fixed it locally by changing the statement in `Google::Cloud::PubSub::V1::Publisher::Credentials` to:

```
            SCOPE = [
              "https://www.googleapis.com/auth/cloud-platform",
              "https://www.googleapis.com/auth/pubsub"
            ]
```


It looks like this problem will also apply to `Google::Auth::Credentials.env_vars` and `Google::Auth::Credentials.paths`. These methods will also not return the values set on instance variables of the `Google::Cloud::PubSub::V1::Publisher::Credentials` subclass.

The next step is to try to understand why the change was made in the Credentials subclasses in [gapic-generator-ruby](https://github.com/googleapis/gapic-generator-ruby). The commit message for [the change](https://github.com/googleapis/gapic-generator-ruby/commit/2dd6ff3153adb8260e739a7ce28a6ee51be03d8e#diff-0d4c9bdc29cb7a773d30cfc52e5d2ec1a0aa15b46792e234a4d5b3e43a8a747f) says simply "Use methods to set cloud Credentials values".

**Comment by quartzmo** [comment]
Fixed in [googleauth v0.15.0](https://rubygems.org/gems/googleauth/versions/0.15.0).
